### PR TITLE
Fixes #33174 - Puppet extraction

### PR DESCRIPTION
**Pull request by amirfefer**
Works with https://github.com/theforeman/hammer-cli-foreman-puppet

### CODE REVIEW

File: config/foreman.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@
   # Basic Auth:
   #:default_auth_type: 'Basic_Auth'
   :username: 'admin'
-  #:password: 'example'
+  :password: 'changeme'
```
Comment (nadjaheitmann) [inline config/foreman.yml:14]: You probably do not want to commit this change.

File: lib/hammer_cli_foreman/references.rb [hunk 2]
```diff
@@ -153,14 +153,6 @@ def self.all_parameters(dsl)
...
     end
 
-    def self.puppetclasses(dsl)
-      dsl.build do
-        collection :puppetclasses, _("Puppetclasses"), :numbered => false do
-          custom_field Fields::Reference
-        end
-      end
-    end
-
```
Comment (nadjaheitmann) [inline lib/hammer_cli_foreman/references.rb:163]: I think that method `def self.environments(dsl)` also needs to be removed.

File: config/foreman.yml (hunk 1 above)
Comment (ofedoren) [reply config/foreman.yml:14]: Yes, I'd rather not change this :)

**Review by ofedoren — CHANGES_REQUESTED** [review]
Thanks for the effort, @amirfefer! LGTM, but

simple search says there are few more places to remove, mostly tests:
 - lib/hammer_cli_foreman/references.rb:115
 - lib/hammer_cli_foreman/location.rb:39
 - lib/hammer_cli_foreman/organization.rb:39
 - test/unit/puppet_class_test.rb
 - test/functional/host_test.rb:359
 - test/functional/hostgroup/create_test.rb few tests that should be adjusted/removed
 - test/functional/hostgroup/update_test.rb dtto
 - test/functional/proxy_test.rb looks like all the tests are related to puppet classes import from a proxy
 - test/functional/smart_class_parameter_test.rb
 - test/functional/template_test.rb:144 && test/functional/template_test.rb:158
 - test/unit/helpers/command.rb:20 && test/unit/helpers/command.rb:24
 - test/unit/host_test.rb few tests that should be adjusted/removed
 - test/unit/hostgroup_test.rb dtto
 - test/unit/id_resolver_test.rb:216
 - test/unit/puppet_class_test.rb
 - test/unit/smart_class_parameter_test.rb
 - test/unit/smart_proxy_test.rb:93

**Comment by amirfefer** [comment]
Thanks @ofedoren, @nadjaheitmann !
Tests are 🟢

File: test/unit/hostgroup_test.rb [hunk 3]
```diff
@@ -67,9 +65,9 @@
     let(:cmd) { HammerCLIForeman::Hostgroup::CreateCommand.new("", ctx) }
 
     context "parameters" do
-      it_should_accept "name, parent_id, environment_id, architecture_id, domain_id, puppet_proxy_id, operatingsystem_id and more",
-          ["--name=hostgroup", "--parent-id=1", "--puppet-environment-id=1", "--architecture-id=1", "--domain-id=1", "--puppet-proxy-id=1",
-            "--operatingsystem-id=1", "--medium-id=1", "--partition-table-id=1", "--subnet-id=1", '--puppet-ca-proxy-id=1', '--puppet-class-ids=1,2', '--root-password=foreman']
+      it_should_accept "name, parent_id, environment_id, architecture_id, domain_id, operatingsystem_id and more",
```
Comment (ofedoren) [inline test/unit/hostgroup_test.rb:68]: environment_id should be also dropped here.

File: test/unit/hostgroup_test.rb [hunk 4]
```diff
@@ -81,9 +79,9 @@
     let(:cmd) { HammerCLIForeman::Hostgroup::UpdateCommand.new("", ctx) }
 
     context "parameters" do
-      it_should_accept "name, parent_id, environment_id, architecture_id, domain_id, puppet_proxy_id, operatingsystem_id and more",
-          ["--id=1 --name=hostgroup2 --title=default/hostgroup2", "--parent-id=1", "--puppet-environment-id=1", "--architecture-id=1", "--domain-id=1", "--puppet-proxy-id=1",
-            "--operatingsystem-id=1", "--medium-id=1", "--partition-table-id=1", "--subnet-id=1", '--puppet-ca-proxy-id=1', '--puppet-class-ids=1,2', '--root-password=foreman']
+      it_should_accept "name, parent_id, environment_id, architecture_id, domain_id, operatingsystem_id and more",
```
Comment (ofedoren) [inline test/unit/hostgroup_test.rb:82]: environment_id should be also dropped here as well.

**Review by ofedoren — CHANGES_REQUESTED** [review]
Great work, @amirfefer! Just few leftovers and I'll merge it :)

Can you please also create a redmine issue and link it here?

**Comment by amirfefer** [comment]
Thanks @ofedoren ! 
done

**Review by ofedoren — APPROVED** [review]
Thanks, @amirfefer!

Merging now :rocket:

**Comment by ekohl** [comment]
This broke our nightly pipelines. That's not something we do during stabilization week. This means we must postpone branching until we get hammer-cli-foreman-puppet released, packaged and supported in our installer.

**Comment by ofedoren** [comment]
@ekohl, can I see logs related to hammer breaking the pipelines? It's possible to restore few places for some dependencies to be available. In this way we wouldn't bring puppet functionallity in hammer-cli-foreman, but it will allow other plugins to have some time to adjust to the extraction.

**Comment by ekohl** [comment]
Oh I should have linked https://community.theforeman.org/t/foreman-nightly-rpm-pipeline-1136-failed/24673.

**Pull request merged**